.github/workflows/run_mobile_comparative_benchmark.yml

```diff
@@ -164,3 +179,110 @@ jobs:
           RESULTS_GCS_DIR: ${{ env.BENCHMARK_GCS_DIR }}/${{ env.TARGET_DEVICE }}-results
         run: |
           gcloud storage cp "${IREE_RESULTS_FILE}" "${RESULTS_GCS_DIR}/"
+
+
+  compile_models_for_android:
+    needs: [setup, build_iree]
+    runs-on:
+      - self-hosted  # must come first
+      - runner-group=${{ needs.setup.outputs.runner-group }}
+      - environment=prod
+      - cpu
+      - os-family=Linux
+    env:
+      BENCHMARK_ARTIFACTS_DIR: android-benchmark-artifacts-dir
+      TARGET_DEVICE: pixel-6-pro
+      IREE_INSTALL_DIR: ${{ needs.build_iree.outputs.iree-install-dir }}
+      IREE_INSTALL_DIR_ARCHIVE: ${{ needs.build_iree.outputs.iree-install-dir-archive }}
+      IREE_INSTALL_DIR_GCS_ARTIFACT: ${{ needs.build_iree.outputs.iree-install-dir-gcs-artifact }}
+      BENCHMARK_GCS_DIR: ${{ needs.setup.outputs.benchmark-gcs-dir }}
+    outputs:
+      benchmark-artifacts-dir: ${{ env.BENCHMARK_ARTIFACTS_DIR }}
+      benchmark-artifacts-dir-archive: ${{ steps.archive.outputs.benchmark-artifacts-dir-archive }}
+      benchmark-artifacts-dir-gcs-artifact: ${{ steps.upload.outputs.benchmark-artifacts-dir-gcs-artifact }}
+    steps:
+      - name: "Checking out PR repository"
+        uses: actions/checkout@e2f20e631ae6d7dd3b768f56a5d2af784dd54791  # v2.5.0
+      - name: "Downloading and unpacking IREE tools"
+        run: |
+          gcloud storage cp "${IREE_INSTALL_DIR_GCS_ARTIFACT}" "${IREE_INSTALL_DIR_ARCHIVE}"
+          tar -xvf "${IREE_INSTALL_DIR_ARCHIVE}"
+      - name: "Compiling workloads"
+        id: compile
+        env:
+          IREE_COMPILE_PATH: ${{ env.IREE_INSTALL_DIR }}/bin/iree-compile
+          OOBI_TEMP_DIR: temp
+        run: |
+          mkdir "${BENCHMARK_ARTIFACTS_DIR}"
+          mkdir "${OOBI_TEMP_DIR}"
+
+          docker run --mount="type=bind,src="${PWD}",target=/work" --workdir="/work" \
+            --env IREE_COMPILE_PATH="${IREE_COMPILE_PATH}" \
+            --env OOBI_TEMP_DIR="${OOBI_TEMP_DIR}" \
+            "gcr.io/iree-oss/openxla-benchmark/android@sha256:34f140fcf806f64f5d0492dfc5af774ea440406264cd68c0405e23a69cbe6d93" \
+              ./experimental/iree/compile_workloads.sh \
+                  "${TARGET_DEVICE}" \
+                  "${BENCHMARK_ARTIFACTS_DIR}"
+      - name: "Creating archives"
```

Comment:
It looks like we take a lot of time to compress the VMFB (27mins) then uncompress on the phone.

I think it might be faster to archive without compression (e.g. `tar -cvf`)

In IREE we actually don't archive the VMFBs but upload and download them individually (`gcloud storage cp -r individually`)

Reply:
Thanks, I removed compression and it reduced the archive step to 30 seconds!